mev-build-rs/src/payload/builder.rs

```diff
@@ -95,10 +96,26 @@ fn append_payment<Client: StateProviderFactory>(
 
     let signer_account = db.load_cache_account(signer.address())?;
     let nonce = signer_account.account_info().map(|info| info.nonce).unwrap_or_default();
+
+    let proposer_fee_recipient_account = db.load_cache_account(config.proposer_fee_recipient)?;
+    let is_empty_code_hash =
+        proposer_fee_recipient_account.account_info().unwrap_or_default().is_empty_code_hash();
+
+    // If the proposer (fee recipient) has associated code, we hardcode 250_000 for gas payment.
+    // If it is a regular EOA, we send the base gas 21_000.
+    let gas_limit = if is_empty_code_hash { BASE_TX_GAS_LIMIT } else { 250_000 };
```

Comment:
would suggest putting another constant next to `BASE_TX_GAS_LIMIT` above

```suggestion
    const PAYMENT_TO_CONTRACT_GAS_LIMIT: u64 = 250_000;
    // ...
    let gas_limit = if is_empty_code_hash { BASE_TX_GAS_LIMIT } else { PAYMENT_TO_CONTRACT_GAS_LIMIT };
```